src/libraries/System.Security.Cryptography.Encoding/tests/PemEncodingTests.cs

```diff
@@ -0,0 +1,548 @@
+            Assert.Equal(content, content[fields.Location]);
+            Assert.Equal(21..21, fields.Base64Data);
+            Assert.Equal(0, fields.DecodedDataLength);
+        }
+
+        [Fact]
+        public static void TryFind_True_EmptyContent_ManyLinesOfWhitespace()
+        {
+            string content = "-----BEGIN EMPTY-----\n\t\n\t\n\t  \n-----END EMPTY-----";
+            Assert.True(PemEncoding.TryFind(content, out PemFields fields));
+            Assert.Equal("EMPTY", content[fields.Label]);
+            Assert.Equal(content, content[fields.Location]);
+            Assert.Equal(30..30, fields.Base64Data);
+            Assert.Equal(0, fields.DecodedDataLength);
+        }
+
+        [Theory]
+        [InlineData("CERTIFICATE")]
+        [InlineData("X509 CRL")]
+        [InlineData("PKCS7")]
+        [InlineData("PRIVATE KEY")]
+        [InlineData("RSA PRIVATE KEY")]
+        public static void TryFind_True_CommonLabels(string label)
+        {
+            string content = $"-----BEGIN {label}-----\nZm9v\n-----END {label}-----";
+            Assert.True(PemEncoding.TryFind(content, out PemFields fields));
+            Assert.Equal(label, content[fields.Label]);
+        }
+
+        [Fact]
+        public static void TryFind_True_LabelCharacterBoundaries()
+        {
+            string content = $"-----BEGIN !PANIC~~~-----\nAHHH\n-----END !PANIC~~~-----";
+            Assert.True(PemEncoding.TryFind(content, out PemFields fields));
+            Assert.Equal("!PANIC~~~", content[fields.Label]);
+        }
+
+        [Fact]
+        public static void TryFind_True_MultiPem()
+        {
+            string content = @"
+-----BEGIN EC PARAMETERS-----
+BgUrgQQACg==
+-----END EC PARAMETERS-----
+-----BEGIN EC PRIVATE KEY-----
+MHQCAQEEIIpP2qP/mGWDAojQDNrNfUHwYGNPKeO6VLt+POJeCJ3OoAcGBSuBBAAK
+oUQDQgAEeDThNbdvTkptgvfNOpETlKBcWDUKs9IcQ/RaFeBntqt+6J875A79YhmD
+D7ofwIDcVqzOJQDhSN54EQ17CFQiwg==
+-----END EC PRIVATE KEY-----
+";
+            ReadOnlySpan<char> pem = content;
+            List<string> labels = new List<string>();
+            while (PemEncoding.TryFind(pem, out PemFields fields))
+            {
+                labels.Add(pem[fields.Label].ToString());
+                pem = pem[fields.Location.End..];
+            }
+
+            Assert.Equal(new string[] { "EC PARAMETERS", "EC PRIVATE KEY" }, labels);
+        }
+
+        [Fact]
+        public static void TryFind_True_FindsPemAfterPemWithInvalidBase64()
+        {
+            string content = @"
+-----BEGIN TEST-----
+$$$$
+-----END TEST-----
+-----BEGIN TEST2-----
+Zm9v
+-----END TEST2-----";
+            Assert.True(PemEncoding.TryFind(content, out PemFields fields));
+            Assert.Equal("TEST2", content[fields.Label]);
+            Assert.Equal("Zm9v", content[fields.Base64Data]);
+        }
+
+        [Fact]
+        public static void TryFind_True_FindsPemAfterPemWithInvalidLabel()
+        {
+            string content = @"
+-----BEGIN ------
```

Comment:
Probably worth a comment that this is an invalid label because there are too many hyphens after the space.